packages/devkit/src/generators/project-configuration.spec.ts

```diff
@@ -24,353 +40,519 @@ const baseTestProjectConfig: ProjectConfiguration = {
 describe('project configuration', () => {
   let tree: Tree;
 
-  beforeEach(() => {
-    tree = createTreeWithEmptyWorkspace();
-  });
+  describe('workspace v1', () => {
+    beforeEach(() => {
+      tree = createTreeWithEmptyWorkspace(1);
+    });
 
-  describe('readProjectConfiguration', () => {
-    it('should get info from workspace.json', () => {
-      updateJson(tree, getWorkspacePath(tree), (json) => {
-        json.projects['proj1'] = {
+    describe('readProjectConfiguration', () => {
+      it('should get info from workspace.json', () => {
+        updateJson(tree, getWorkspacePath(tree), (json) => {
+          json.projects['proj1'] = {
+            root: 'proj1',
+          };
+          return json;
+        });
+
+        const config = readProjectConfiguration(tree, 'proj1');
+        expect(config).toEqual({
           root: 'proj1',
-        };
-        return json;
+        });
       });
 
-      const config = readProjectConfiguration(tree, 'proj1');
-      expect(config).toEqual({
-        root: 'proj1',
+      it('should get info from nx.json', () => {
+        updateJson(tree, getWorkspacePath(tree), (json) => {
+          json.projects['proj1'] = {
+            root: 'proj1',
+          };
+          return json;
+        });
+        updateJson(tree, 'nx.json', (json) => {
+          json.projects['proj1'] = {
+            tags: ['tag1'],
+          };
+          return json;
+        });
+
+        const config = readProjectConfiguration(tree, 'proj1');
+        expect(config).toEqual({
+          root: 'proj1',
+          tags: ['tag1'],
+        });
       });
-    });
 
-    it('should get info from nx.json', () => {
-      updateJson(tree, getWorkspacePath(tree), (json) => {
-        json.projects['proj1'] = {
+      it('should should not fail if projects is not defined in nx.json', () => {
+        updateJson(tree, getWorkspacePath(tree), (json) => {
+          json.projects['proj1'] = {
+            root: 'proj1',
+          };
+          return json;
+        });
+        updateJson(tree, 'nx.json', (json) => {
+          delete json.projects;
+          return json;
+        });
+
+        const config = readProjectConfiguration(tree, 'proj1');
+        expect(config).toEqual({
           root: 'proj1',
-        };
-        return json;
+        });
       });
-      updateJson(tree, 'nx.json', (json) => {
-        json.projects['proj1'] = {
-          tags: ['tag1'],
-        };
-        return json;
+    });
+
+    describe('addProjectConfiguration', () => {
+      it('should throw when standalone is true', () => {
```

Comment:
It seems like this didn't throw before... why does it throw now?

Reply:
Before, nothing in `addProjectConfiguration` was making the check to disallow standalone projects in v1 workspaces. That check was happening when calling `formatFiles` which was the one calling `reformattedWorkspaceJsonOrNull` which calls `toOldFormatOrNull` when the workspace is v1. That last function is the one making the check.

With the changes in this PR, as part of any function to add/update the configuration, the `reformattedWorkspaceJsonOrNull` is called so the configuration is written in the right format and therefore the check happens.